File: app/src/main/java/com/iavariav/kbmonline/ui/atasan/presenter/AprovalPresenter.java
package com.iavariav.kbmonline.ui.atasan.presenter;

import android.content.Context;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.Toast;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.iavariav.kbmonline.model.PemesananModel;
import com.iavariav.kbmonline.rest.ApiConfig;
import com.iavariav.kbmonline.rest.ApiService;
import com.iavariav.kbmonline.ui.atasan.adapter.AtasanAprovalAdapter;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class AprovalPresenter {
    private ArrayList<PemesananModel> PemesananModels;
    private AtasanAprovalAdapter atasanAprovalAdapter;

    public void getDatas(final Context context, final RecyclerView rv, final LinearLayout div) {
        PemesananModels = new ArrayList<>();
        ApiService apiService = ApiConfig.getApiService();
        apiService.getAllData("getAllDataPemesanan")
                .enqueue(new Callback<ArrayList<PemesananModel>>() {
                    @Override
                    public void onResponse(Call<ArrayList<PemesananModel>> call, Response<ArrayList<PemesananModel>> response) {
                        if (response.isSuccessful()){
                            PemesananModels = response.body();
                            atasanAprovalAdapter = new AtasanAprovalAdapter(context, PemesananModels);
                            rv.setLayoutManager(new LinearLayoutManager(context));
                            rv.setAdapter(atasanAprovalAdapter);
                            atasanAprovalAdapter.notifyDataSetChanged();

                            if (PemesananModels.isEmpty()) {
                                rv.setVisibility(View.GONE);
                                div.setVisibility(View.VISIBLE);
                            } else {
                                rv.setVisibility(View.VISIBLE);
                                div.setVisibility(View.GONE);
                            }
                        }
                    }

                    @Override
                    public void onFailure(Call<ArrayList<PemesananModel>> call, Throwable t) {

                        Toast.makeText(context, "" + t.getMessage(), Toast.LENGTH_SHORT).show();
                    }
                });
    }
}

File: app/src/main/java/com/iavariav/kbmonline/ui/atasan/presenter/AprovalPresenter.kt
package com.iavariav.kbmonline.ui.atasan.presenter

import android.view.View
import android.widget.LinearLayout
import android.widget.Toast
import androidx.fragment.app.FragmentActivity

import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.iavariav.kbmonline.model.PemesananModel
import com.iavariav.kbmonline.rest.ApiConfig
import com.iavariav.kbmonline.ui.atasan.adapter.AtasanAprovalAdapter

import java.util.ArrayList

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class AprovalPresenter {
    private var PemesananModels: ArrayList<PemesananModel>? = null
    private var atasanAprovalAdapter: AtasanAprovalAdapter? = null

    fun getDatas(context: FragmentActivity?, rv: RecyclerView?, div: LinearLayout?) {
        PemesananModels = ArrayList()
        val apiService = ApiConfig.apiService
        apiService.getAllData("getAllDataPemesanan")
                .enqueue(object : Callback<ArrayList<PemesananModel>> {
                    override fun onResponse(call: Call<ArrayList<PemesananModel>>, response: Response<ArrayList<PemesananModel>>) {
                        if (response.isSuccessful) {
                            PemesananModels = response.body()
                            atasanAprovalAdapter = AtasanAprovalAdapter(context, PemesananModels!!)
                            rv?.layoutManager = LinearLayoutManager(context)
                            rv?.adapter = atasanAprovalAdapter
                            atasanAprovalAdapter!!.notifyDataSetChanged()

                            if (PemesananModels!!.isEmpty()) {
                                rv?.visibility = View.GONE
                                div?.visibility = View.VISIBLE
                            } else {
                                rv?.visibility = View.VISIBLE
                                div?.visibility = View.GONE
                            }
                        }
                    }

                    override fun onFailure(call: Call<ArrayList<PemesananModel>>, t: Throwable) {

                        Toast.makeText(context, "" + t.message, Toast.LENGTH_SHORT).show()
                    }
                })
    }
}
